contracts/extensions/token/erc20/UnitToken.sol

```diff
@@ -0,0 +1,237 @@
+        dao = _dao;
+    }
+
+    /**
+     * @dev Returns the name of the token.
+     */
+    function name() public view virtual returns (string memory) {
+        return "Unit Token";
+    }
+
+    /**
+     * @dev Returns the symbol of the token, usually a shorter version of the
+     * name.
+     */
+    function symbol() public view virtual returns (string memory) {
+        return "UNIT";
+    }
+
+    /**
+     * @dev Returns the number of decimals used to get its user representation.
+     * For example, if `decimals` equals `2`, a balance of `505` tokens should
+     * be displayed to a user as `5,05` (`505 / 10 ** 2`).
+     */
+    function decimals() public view virtual returns (uint8) {
+        return 18;
+    }
+
+    /**
+     * @dev Returns the amount of tokens in existence.
+     */
+    function totalSupply() public view override returns (uint256) {
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+        return bank.balanceOf(TOTAL, UNITS);
+    }
+
+    /**
+     * @dev Returns the amount of tokens owned by `account`.
+     */
+    function balanceOf(address account) public view override returns (uint256) {
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+        return bank.balanceOf(account, UNITS);
+    }
+
+    /**
+     * @dev Returns the remaining number of tokens that `spender` will be
+     * allowed to spend on behalf of `owner` through {transferFrom}. This is
+     * zero by default.
+     *
+     * This value changes when {approve} or {transferFrom} are called.
+     */
+    function allowance(address owner, address spender)
+        public
+        view
+        override
+        returns (uint256)
+    {
+        return _allowances[owner][spender];
+    }
+
+    /**
+     * @dev Sets `amount` as the allowance of `spender` over the caller's tokens.
+     *
+     * Returns a boolean value indicating whether the operation succeeded.
+     *
+     * IMPORTANT: Beware that changing an allowance with this method brings the risk
+     * that someone may use both the old and the new allowance by unfortunate
+     * transaction ordering. One possible solution to mitigate this race
+     * condition is to first reduce the spender's allowance to 0 and set the
+     * desired value afterwards:
+     * https://github.com/ethereum/EIPs/issues/20#issuecomment-263524729
+     *
+     * Emits an {Approval} event.
+     */
+    function approve(address spender, uint256 amount)
+        public
+        override
+        returns (bool)
+    {
+        address senderAddr = dao.getAddressIfDelegated(msg.sender);
+        require(
+            senderAddr != address(0),
+            "ERC20: approve from the zero address"
+        );
+        require(spender != address(0), "ERC20: approve to the zero address");
+        require(dao.isMember(senderAddr), "sender is not a member");
+
+        BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
+        require(
```

Comment:
AFAIK, you can give allowance even if you don't have all the tokens yet. It is useful when you want to give access to future tokens as well (if you know you'll get more in the future)